### PR TITLE
add E4S Product Dictionary

### PR DESCRIPTION
**Pull request by shahzebsiddiqui**
@wspear @sameershende @eugeneswalker @maherou @tgamblin can you take a look this PR this will address #11 and provide clarity on list of E4S packages since we have multiple sources. 

This will take the README.md from https://github.com/E4S-Project/e4s-ci-badges/blob/main/README.md and stick in this repo with some changes.

You can see the E4S Product Dictionary in my branch: https://github.com/shahzebsiddiqui/e4s/blob/e4s_product_dictionary/E4S_Products.md 

We need to make sure all of these pages point to a single source for E4S products. We have some members responsible for these pages 

Link | Assignee
---- | ----------
https://confluence.exascaleproject.org/display/HISD/SI+Build+and+Test+Dashboard |  @tldahlgren
https://e4s-project.github.io/DocPortal.html | @wspear 
https://confluence.exascaleproject.org/display/1ST/ECP+ST+Product+Readiness | @maherou @rmadamson @shahzebsiddiqui 
E4S spack.yaml distribution (https://github.com/E4S-Project/e4s/tree/master/environments/21.05) | @eugeneswalker 

I would request everyone to review the list and make sure I have not missed any packages part of https://confluence.exascaleproject.org/display/HISD/SI+Build+and+Test+Dashboard and https://confluence.exascaleproject.org/display/1ST/ECP+ST+Product+Readiness. There was some confusion on some packages that are part of `xsdk` which was brought into the spack.yaml release of 21.05. Some of this is documented in #11 and slack message https://exascaleproject.slack.com/archives/C021ZPW3LTS/p1628693288017900.

I would like to see this product list finalized as part of the next release (21.08). @eugeneswalker has shared link to the upcoming spack.yaml (https://github.com/spack/spack-configs/blob/main/UOREGON/E4S-Develop/spack-ubuntu20.04-x86_64.yaml) some of which are questionable. For Software Integration team, we are expected to report bugs and any issues in https://confluence.exascaleproject.org/display/1ST/ECP+ST+Product+Readiness for Frontier and Spock system so we need to make sure the table in confluence is in sync with the E4S Product Dictionary so we can report issues properly in the correct row+column. 

Once this PR is merged, i would like to incorporate these changes in https://github.com/spack/spack/pull/23382 which will report packages by `e4s` and `ecp` tag.

There is a high degree of overlap between e4s list and xsdk list. I am not sure if it makes sense to document this here or this should be left to xsdk team to document in their repos (https://github.com/xsdk-project). If we decide to document here i need to know list of xsdk packages perhaps we can have a separate `.md` file or same file but two different tables.

### CODE REVIEW

**Comment by eugeneswalker** [comment]
I'm looking at the rendered product dictionary: https://github.com/shahzebsiddiqui/e4s/blob/e4s_product_dictionary/E4S_Products.md

and there are multiple broken images??

**Comment by maherou** [comment]
I like the new version of README.md and the E4S_Products.md file.  As much as possible, let’s make sure the E4S_Products.md file requires minimal manual updates.  Otherwise, I think it looks good.

We should also look at how this information could be linked to the DocPortal, or perhaps integrated into it in some way.

**Comment by shahzebsiddiqui** [comment]
@maherou  this PR #20 will help us out, it will automatically check urls since there are going to be many of them. I have used this action in several of my projects and it is quite handy for documentation with external links that get broken over time.

**Comment by shahzebsiddiqui** [comment]
question for @maherou are there any packages missing from this list. Or ones that should be removed. Please see https://github.com/shahzebsiddiqui/e4s/blob/e4s_product_dictionary/E4S_Products.md for rendered output

**Comment by maherou** [comment]
I don’t know if this list is accurate.  We need a deterministic approach to defining the list.

**Comment by shahzebsiddiqui** [comment]
okay i have added few more packages like `argobots`, `axom`, `llvm`, and `pdt` to the list see https://github.com/E4S-Project/e4s/pull/19/commits/4a3108ce103e6fb42b2647221dccdc72ea0ee474. Are there any other corrections to this list?

**Comment by eugeneswalker** [comment]
What about the packages in the product dictionary which *do not* have Spack packages? My understanding up to this point has been that to be included in E4S there must be a Spack package.

**Comment by shahzebsiddiqui** [comment]
> What about the packages in the product dictionary which _do not_ have Spack packages? My understanding up to this point has been that to be included in E4S there must be a Spack package.

I think we should allow user to provide request to be added to E4S in the product form we have option to select this.

Screen Shot 2021-08-11 at 6 55 45 PM

According to community policy https://e4s-project.github.io/policies.html **P1** assumes there must be a spack package though it is my understanding that teams will attempt to work through the community policy. Or do we reject the request at the time user creates an issue. Question for @maherou, how strictly do we follow the community policy and what actions are taken to enforce teams adhere to policy. There should be some data to provide compatibility for each product with community policy.

**Comment by shahzebsiddiqui** [comment]
i have removed the `CI Badge` column now @wspear in https://github.com/E4S-Project/e4s/pull/19/commits/439600e8802b32859a3f3f9aae0072492a866ee0

**Comment by rmadamson** [comment]
I don't think we have a mapping of E4S project <-> spack package(s) defined anywhere publicly yet, so this seems like a good thing to do.

I wanted to echo @maherou 's comments regarding generating this list in a deterministic way.  Do we intend this to be the authoritative source of spack packages that are in E4S, or is that better captured in https://e4s-project.github.io/DocPortal.html somehow where the product teams themselves maintain the spack packages they have?

File: E4S_Products.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+| [aml](https://xgitlab.cels.anl.gov/argo/aml)          | | | ![spack](https://img.shields.io/spack/v/aml) | [aml](https://github.com/spack/spack/tree/develop/var/spack/repos/builtin/packages/aml) |
+| [amrex](https://github.com/AMReX-Codes/amrex)         |  ![release](https://img.shields.io/github/v/release/AMReX-Codes/amrex) | ![release-date](https://img.shields.io/github/release-date/AMReX-Codes/amrex) | ![spack](https://img.shields.io/spack/v/amrex) |  [amrex](https://github.com/spack/spack/tree/develop/var/spack/repos/builtin/packages/amrex) |
+| [arborx-x](https://github.com/arborx/ArborX)          | ![release](https://img.shields.io/github/v/release/arborx/ArborX) | ![release-date](https://img.shields.io/github/release-date/arborx/ArborX) | ![spack](https://img.shields.io/spack/v/arborx) | [arborx](https://github.com/spack/spack/tree/develop/var/spack/repos/builtin/packages/arborx) | [aprokop](https://github.com/aprokop)
```
Comment (eugeneswalker) [inline E4S_Products.md:11]: minor typo: `arborx-x` should be `arborx`

**Review by eugeneswalker — CHANGES_REQUESTED** [review]

File: E4S_Products.md (hunk 1 above)
Comment (shahzebsiddiqui) [reply E4S_Products.md]: good catch. This is now fixed in https://github.com/E4S-Project/e4s/pull/19/commits/5aed7d88262c6262dd519f2595fcbebf60364306

**Review by eugeneswalker — APPROVED** [review]

**Comment by shahzebsiddiqui** [comment]
@sameershende @maherou would you mind reviewing this PR and let me know if you would like any further corrections. I tried my best to put as many maintainers for the products based on `maintainers` property in spack though some packages didnt have maintainers and I would rather let the teams update the missing entry. I think the `maintainer` column can be used to contact the team directly for any E4S support.

**Comment by eugeneswalker** [comment]
This is missing PDT. It should be a Core E4S package.

**Comment by shahzebsiddiqui** [comment]
@sameershende i have removed all the products not part of e4s in https://github.com/E4S-Project/e4s/pull/19/commits/2316deb5654d0a18abc9f9f19d42ae9cf902301b. If there are any that need to be added, will add them as subsequent PR. I am resolving this and merging this.

**Pull request merged**